crates/starknet-types-core/src/felt/qm31.rs

```
@@ -0,0 +1,447 @@
+    /// # Safety
+    /// If the value is zero will panic.
+    pub fn inverse(&self) -> QM31Felt {
+        assert_ne!(*self, Self::ZERO, "Zero is not an invertible number");
+
+        let coordinates = self.as_raw();
+
+        let b2_r = (coordinates[2] * coordinates[2] + STWO_PRIME * STWO_PRIME
+            - coordinates[3] * coordinates[3])
+            % STWO_PRIME;
+        let b2_i = (2 * coordinates[2] * coordinates[3]) % STWO_PRIME;
+
+        let denom_r = (coordinates[0] * coordinates[0] + STWO_PRIME * STWO_PRIME
+            - coordinates[1] * coordinates[1]
+            + 2 * STWO_PRIME
+            - 2 * b2_r
+            + b2_i)
+            % STWO_PRIME;
+        let denom_i =
+            (2 * coordinates[0] * coordinates[1] + 3 * STWO_PRIME - 2 * b2_i - b2_r) % STWO_PRIME;
+
+        let denom_norm_squared = (denom_r * denom_r + denom_i * denom_i) % STWO_PRIME;
+        let denom_norm_inverse_squared = Self::pow2147483645(denom_norm_squared);
+
+        let denom_inverse_r = (denom_r * denom_norm_inverse_squared) % STWO_PRIME;
+        let denom_inverse_i = ((STWO_PRIME - denom_i) * denom_norm_inverse_squared) % STWO_PRIME;
+
+        Self::from_raw([
+            coordinates[0] * denom_inverse_r + STWO_PRIME * STWO_PRIME
+                - coordinates[1] * denom_inverse_i,
+            coordinates[0] * denom_inverse_i + coordinates[1] * denom_inverse_r,
+            coordinates[3] * denom_inverse_i + STWO_PRIME * STWO_PRIME
+                - coordinates[2] * denom_inverse_r,
+            2 * STWO_PRIME * STWO_PRIME
+                - coordinates[2] * denom_inverse_i
+                - coordinates[3] * denom_inverse_r,
+        ])
+    }
+
+    /// Computes the division of two QM31 elements in reduced form.
+    /// Returns an error if the input is zero.
+    /// # Safety
+    /// Will panic if the rhs value is equal to zero.
+    pub fn div(&self, rhs: &QM31Felt) -> Result<QM31Felt, QM31Error> {
```

The signature of the function contains a result, but it will panic instead of returning an error. Can we remove the panic and keep the result?